pkg/scheduler/framework/plugins/podtopologyspread/common.go

```diff
@@ -66,19 +68,27 @@ func nodeLabelsMatchSpreadConstraints(nodeLabels map[string]string, constraints
 	return true
 }
 
-func filterTopologySpreadConstraints(constraints []v1.TopologySpreadConstraint, action v1.UnsatisfiableConstraintAction) ([]topologySpreadConstraint, error) {
+func filterTopologySpreadConstraints(constraints []v1.TopologySpreadConstraint, action v1.UnsatisfiableConstraintAction, feature feature.Features) ([]topologySpreadConstraint, error) {
 	var result []topologySpreadConstraint
 	for _, c := range constraints {
 		if c.WhenUnsatisfiable == action {
 			selector, err := metav1.LabelSelectorAsSelector(c.LabelSelector)
 			if err != nil {
 				return nil, err
 			}
-			result = append(result, topologySpreadConstraint{
+			tsc := topologySpreadConstraint{
 				MaxSkew:     c.MaxSkew,
 				TopologyKey: c.TopologyKey,
 				Selector:    selector,
-			})
+			}
+			if feature.EnableMinDomainsInPodTopologySpread {
+				// The default value of MinDomains is 0.
+				tsc.MinDomains = 0
+				if c.MinDomains == nil {
+					tsc.MinDomains = *c.MinDomains
```

Comment:
Bug here?  c.MinDomains is nil.  It is better to add UT for this.
`c.MinDomains != nil` ?

Reply:
Thanks for nice catching... 🙏 
Exactly, I added the unit test for this.